internal/framework/runnables/job.go

```diff
@@ -0,0 +1,52 @@
+package runnables
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+// JobConfig is the configuration for a job.
+type JobConfig struct {
+	// Worker is the function that will be run for every job iteration.
+	Worker func(context.Context)
+	// Logger is the logger.
+	Logger logr.Logger
+	// Period defines the period of the job. The job will run every Period.
+	Period time.Duration
+}
+
+// Job periodically runs a worker function.
+type Job struct {
+	cfg JobConfig
+}
+
+// NewJob creates a new job.
+func NewJob(cfg JobConfig) *Job {
+	return &Job{
+		cfg: cfg,
+	}
+}
+
+// Start starts the job.
+// Implements controller-runtime manager.Runnable
+func (j *Job) Start(ctx context.Context) error {
+	j.cfg.Logger.Info("Starting job")
+
+	const (
+		// 10 min jitter is enough per telemetry destination recommendation
```

Comment:
We should remove comments related to telemetry. On a related note, maybe the jitter should be configurable if this job is meant to be generic.

Reply:
Yep, missed that one.

Reply:
+1 to make jitter configurable. this way, we can move the comment about DataFabric recommended jitter to 
```
	return &runnables.Leader{
		Runnable: runnables.NewCronJob(
			runnables.CronJobConfig{
				Worker: worker,
				Logger: logger,
				Period: cfg.TelemetryReportPeriod,
			},
		),
	}
```

Reply:
Yeah sorry, missed that part of the comment.